src/containers/BooksList.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import Book from '../components/Book';
+
+import { connect } from 'react-redux';
+import { removeBook, changeFilter } from '../actions';
+
+import CategoryFilter from '../components/CategoryFilter';
+
+const getVisibleBooks = (books, currentFilter) => {

```

Comment:
We can do away from having a `getVisibleBooks` function here by just filtering the `books` in the `mapStateToProps` itself:

```javascript
const mapStateToProps = state => ({
  books: state.books.filter(({category}) => category === state.filter);
  currentFilter: state.filter,
})
```

**Note: We don't need any conditional for when `filter === ''` because if it is `''` then no book will be returned anyway**

Reply:
Honestly, I am confused now with all that stuff on getting dummy books and how not to use this condition. If filter is empty I need to return all the books, not none of them, no?